### DISCUSSION TITLE
Enum template

### DISCUSSION THREAD

**Discussion by aannnn24** [microsoft/typespec#3282] [General]
Hi,

Can enums be used together with templates?

I have the following template model:
```ts
model Address<T> { type: AdressType }
enum AdressType { Home, Work, Other }
```
When T is a Person I would like type possible values to be set to Home/Work and when T is Company, set to Other. Is there a conditional operator or template behaviour to do it?

Thank you

**Comment by timotheeguerin** [comment]
We do not have conditional templates like TypeScript does today
